app/src/main/java/com/example/playlistmaker/components/TrackList.kt

```diff
@@ -0,0 +1,192 @@
+            verticalArrangement = Arrangement.Center,
+        ) {
+            trackName?.let {
+                Text(
+                    text = it,
+                    style = AppTheme.typography.body,
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis
+                )
+            }
+            Row(horizontalArrangement = Arrangement.Start) {
+                trackArtist?.let {
+                    Text(
+                        text = it,
+                        style = AppTheme.typography.body3,
+                        maxLines = 1,
+                        overflow = TextOverflow.Ellipsis,
+                    )
+                }
+                Icon(
+                    painter = painterResource(R.drawable.ellipse_icon13),
+                    contentDescription = "",
+                    tint = AppTheme.colors.additionColor
+                )
+                Text(
+                    text = trackDuration,
+                    style = AppTheme.typography.body3,
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis
+                )
+            }
+        }
+
+        Icon(
+            painter = painterResource(R.drawable.arrow_forward_icon),
+            contentDescription = stringResource(R.string.play_btn_desc),
+            tint = AppTheme.colors.additionColor,
+        )
+    }
+}
+
+@Composable
+fun TrackSmallCover(url: String?) {
+    AsyncImage(
+        model = url,
+        contentDescription = stringResource(R.string.cover_ph_desc),
+        placeholder = painterResource(R.drawable.cover_placeholder),
+        contentScale = ContentScale.Crop,
+        modifier = Modifier
+            .size(45.dp)
+            .clip(RoundedCornerShape(2.dp)),
+        error = painterResource(R.drawable.cover_placeholder),
+    )
+}
+
+@Composable
+fun TrackListHistory(
+    tracks: List<ListTrackInfo>,
+    onTrackClick: (trackId: Int) -> Unit,
+    onButtonClick: () -> Unit
+) {
+    LazyColumn(
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier.padding(top = 16.dp)
+    ) {
+        items(tracks) { track ->
+            TrackItem(
+                track.trackId,
+                track.trackName,
+                track.artistName,
+                track.artworkUrl,
+                track.trackTime,
+                onTrackClick)
+        }
+        item {
+            Spacer(modifier = Modifier.height(24.dp))
+            Button(title = stringResource(R.string.search_history_clear), onClick = onButtonClick)
+            Spacer(modifier = Modifier.height(60.dp))
+        }
+    }
+}
+
+@Composable
+fun ListOfListTrackInfo(tracks: List<ListTrackInfo>, onTrackClick: (trackId: Int) -> Unit) {
+    LazyColumn(
+        verticalArrangement = Arrangement.Top,
+        modifier = Modifier.padding(top = 16.dp)
+    ) {
+        items(tracks) { track ->
```

Comment:
Для повышения стабильности композабл функций в функцию `items` лучше указывать уникальные `key`:
```suggestion
        items(tracks, key = { it.trackId }, contentType = { Track::class.simpleName }) { track ->
```